variables.tf

```diff
@@ -15,23 +15,56 @@ variable "name" {
 }
 
 # This is required for most resource modules
-variable "resource_group" {
+variable "resource_group_name" {
+  type        = string
+  description = "The resource group where the resources will be deployed."
+}
+
+variable "ai_studio_hub_id" {
+  type        = string
+  default     = null
+  description = "The AI Studio Hub ID for which to create a Project"
+
+  validation {
+    condition     = var.kind != "Project" || var.ai_studio_hub_id != null
+    error_message = "The Hub ID is required when `var.kind` equals 'Project'"
+  }
+}
+
+variable "aiservices" {
   type = object({
-    id   = string
-    name = string
+    ignore                = optional(bool, true)
```

Comment:
what is the purpose for the ignore value? can we leverage validation for this?

Reply:
Simplifying validation to allow `create_new` to be false & properties required for existing resource as null is a cleaner solution. Updating accordingly.